sdk/core/src/services/WalletStorageService.ts

```diff
@@ -204,10 +217,12 @@ export default class WalletStorageService {
       throw new SdkErrorFromCode('COR-0', {}, error)
     }
 
-    try {
-      await this._bloomVaultStorageService.deleteAllCredentials(storageRegion)
-    } catch (error) {
-      throw new SdkErrorFromCode('COR-0', {}, error)
+    if (this._queryBloomVault) {
+      try {
+        await this._bloomVaultStorageService.deleteAllCredentials(storageRegion)
+      } catch (error) {
+        throw new SdkErrorFromCode('COR-0', {}, error)
+      }
```

Comment:
I'd recommend moving this part into separate function to reduce the amount of nested code, before these changes it was 1 nested block - now 2